### PR TITLE
perf(TestComparer): Optimize GC memory allocation.

### PR DESCRIPTION
**Pull request by workgroupengineering**
GitHub Issue (If applicable): closes #
```ini

BenchmarkDotNet=v0.13.1, OS=Windows 10.0.19042.1237 (20H2/October2020Update)
Intel Core i7-7700K CPU 4.20GHz (Kaby Lake), 1 CPU, 8 logical and 4 physical cores
  [Host]     : .NET Framework 4.8 (4.8.4400.0), X64 RyuJIT
  DefaultJob : .NET Framework 4.8 (4.8.4400.0), X64 RyuJIT


```
|   Method |      data |     Mean |   Error |   StdDev |   Median |      Min |      Max | Ratio | RatioSD | Rank |     Gen 0 | Allocated native memory | Native memory leak |     Gen 1 |     Gen 2 |     Allocated |
|--------- |---------- |---------:|--------:|---------:|---------:|---------:|---------:|------:|--------:|-----:|----------:|------------------------:|-------------------:|----------:|----------:|--------------:|
| **Uno(WPF)** | **T (1).png** | **281.9 ms** | **5.57 ms** | **12.91 ms** | **277.5 ms** | **266.6 ms** | **319.9 ms** |  **1.00** |    **0.00** |    **3** | **2000.0000** |           **157,828,212 B** |      **156,796,952 B** | **1000.0000** | **1000.0000** | **179,190,840 B** |
|     Skia | T (1).png | 228.4 ms | 6.94 ms | 20.34 ms | 219.9 ms | 204.7 ms | 286.1 ms |  0.81 |    0.07 |    1 |         - |            45,256,907 B |                  - |         - |         - |       5,461 B |
|   OpenCV | T (1).png | 257.9 ms | 2.14 ms |  1.79 ms | 257.5 ms | 254.7 ms | 262.0 ms |  0.90 |    0.06 |    2 |         - |            45,280,772 B |                  - |         - |         - |             - |
|          |           |          |         |          |          |          |          |       |         |      |           |                         |                    |           |           |               |
| **Uno(WPF)** | **T (2).png** | **293.2 ms** | **5.99 ms** | **17.57 ms** | **288.8 ms** | **266.7 ms** | **337.9 ms** |  **1.00** |    **0.00** |    **3** | **1500.0000** |           **157,828,212 B** |      **156,796,952 B** |  **500.0000** |  **500.0000** | **179,182,372 B** |
|     Skia | T (2).png | 206.0 ms | 1.25 ms |  1.39 ms | 205.6 ms | 204.4 ms | 209.2 ms |  0.69 |    0.03 |    1 |         - |            45,256,907 B |                  - |         - |         - |       5,461 B |
|   OpenCV | T (2).png | 265.2 ms | 5.20 ms |  9.11 ms | 264.0 ms | 254.4 ms | 287.7 ms |  0.90 |    0.06 |    2 |         - |            45,280,772 B |                  - |         - |         - |             - |
|          |           |          |         |          |          |          |          |       |         |      |           |                         |                    |           |           |               |
| **Uno(WPF)** | **T (3).png** | **288.4 ms** | **5.27 ms** |  **8.80 ms** | **287.3 ms** | **274.3 ms** | **314.2 ms** |  **1.00** |    **0.00** |    **3** | **1500.0000** |           **157,828,084 B** |      **156,796,952 B** |  **500.0000** |  **500.0000** | **179,190,252 B** |
|     Skia | T (3).png | 206.8 ms | 1.37 ms |  1.07 ms | 206.9 ms | 204.5 ms | 208.6 ms |  0.73 |    0.02 |    1 |         - |            45,269,838 B |                  - |         - |         - |       5,461 B |
|   OpenCV | T (3).png | 274.4 ms | 1.61 ms |  1.34 ms | 274.5 ms | 271.0 ms | 276.6 ms |  0.96 |    0.03 |    2 |         - |            45,279,868 B |                  - |         - |         - |             - |
|          |           |          |         |          |          |          |          |       |         |      |           |                         |                    |           |           |               |
| **Uno(WPF)** | **T (4).png** | **273.6 ms** | **4.82 ms** |  **4.51 ms** | **274.4 ms** | **265.9 ms** | **282.3 ms** |  **1.00** |    **0.00** |    **3** | **1500.0000** |           **157,828,212 B** |      **156,796,952 B** |  **500.0000** |  **500.0000** | **179,182,372 B** |
|     Skia | T (4).png | 206.3 ms | 1.03 ms |  0.96 ms | 206.6 ms | 204.4 ms | 207.6 ms |  0.75 |    0.01 |    1 |         - |            45,256,907 B |                  - |         - |         - |       5,461 B |
|   OpenCV | T (4).png | 257.2 ms | 1.22 ms |  1.02 ms | 257.3 ms | 254.8 ms | 258.7 ms |  0.94 |    0.02 |    2 |         - |            45,280,772 B |                  - |         - |         - |             - |

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the new behavior?

- Reduces pressure on the GC
- about 28 percent faster

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.TestComparer/Comparer/TestFilesComparer.cs [hunk 1]
```diff
@@ -257,74 +248,90 @@ private IEnumerable<string> EnumerateFiles(string path, string pattern)
...
+			// Adjuts color
+			if (img1.ColorType != SKColorType.Rgba8888)
 			{
-				currentImage[i] = (byte)(currentImage[i] ^ previousImage[i]);
+				using (var old = pixmap1)
+					pixmap1 = old.WithColorType(SKColorType.Rgba8888);
 			}
 
-			if (pixelSize == 4)
+			// Adjuts color
```
Comment (jeromelaban) [inline src/Uno.UI.TestComparer/Comparer/TestFilesComparer.cs:275]:
```suggestion
			// Adjust color
```

File: src/Uno.UI.TestComparer/Comparer/TestFilesComparer.cs [hunk 2]
```diff
@@ -257,74 +248,90 @@ private IEnumerable<string> EnumerateFiles(string path, string pattern)
...
+			using var img2 = SKBitmap.Decode(@"\\?\" + b);
+
+			var pixmap1 = img1.PeekPixels();
+			// Adjuts color
+			if (img1.ColorType != SKColorType.Rgba8888)
 			{
-				currentImage[i] = (byte)(currentImage[i] ^ previousImage[i]);
+				using (var old = pixmap1)
+					pixmap1 = old.WithColorType(SKColorType.Rgba8888);
 			}
```
Comment (jeromelaban) [inline src/Uno.UI.TestComparer/Comparer/TestFilesComparer.cs:273]: Better make a method for this, as it's duplicated below

File: src/Uno.UI.TestComparer/Comparer/TestFilesComparer.cs [hunk 3]
```diff
@@ -257,74 +248,90 @@ private IEnumerable<string> EnumerateFiles(string path, string pattern)
...
 
-			if (pixelSize == 4)
+			// Adjuts color
+			var pixmap2 = img2.PeekPixels();
+			if (img1.ColorType != SKColorType.Rgba8888)
```
Comment (jeromelaban) [inline src/Uno.UI.TestComparer/Comparer/TestFilesComparer.cs:277]: Should have been:
```suggestion
			if (img2.ColorType != SKColorType.Rgba8888)
```

But best in a separate method.

**Comment by jeromelaban** [comment]
Thanks for the contribution! While the change is indeed allocating a bit less, it's not having a significant impact on CI machines (most of the time is spent on disk and network I/O, not on CPU time).

Did you notice a change when running locally ?

**Comment by workgroupengineering** [comment]
You're right, I was just focused on memory usage, not much can be done on network i / o.

After this change fc45648858c295467579402eb13ab40802aa896b, however, I have optimized the use of the cpu by about 30%.

It sure is always better than nothing.

``` ini

BenchmarkDotNet=v0.13.1, OS=Windows 10.0.19042.1237 (20H2/October2020Update)
Intel Core i7-7700K CPU 4.20GHz (Kaby Lake), 1 CPU, 8 logical and 4 physical cores
  [Host]   : .NET Framework 4.8 (4.8.4400.0), X64 RyuJIT
  ShortRun : .NET Framework 4.8 (4.8.4400.0), X64 RyuJIT

Job=ShortRun  InvocationCount=1  IterationCount=3  
LaunchCount=1  UnrollFactor=1  WarmupCount=3  

```
|   Method |                pairs | write |     Mean |    Error |  StdDev |      Min |      Max |   Median | Ratio | Rank |
|--------- |--------------------- |------ |---------:|---------:|--------:|---------:|---------:|---------:|------:|-----:|
| **Uno(WPF)** | **Syste(...)Pair] [53]** | **False** |  **92.50 s** | **14.942 s** | **0.819 s** |  **91.56 s** |  **93.08 s** |  **92.85 s** |  **1.00** |    **2** |
|     Skia | Syste(...)Pair] [53] | False |  83.90 s |  1.619 s | 0.089 s |  83.82 s |  83.99 s |  83.88 s |  0.91 |    1 |
|          |                      |       |          |          |         |          |          |          |       |      |
| **Uno(WPF)** | **Syste(...)Pair] [53]** |  **True** | **133.62 s** | **36.284 s** | **1.989 s** | **131.87 s** | **135.78 s** | **133.20 s** |  **1.00** |    **2** |
|     Skia | Syste(...)Pair] [53] |  True |  87.81 s |  9.523 s | 0.522 s |  87.21 s |  88.19 s |  88.02 s |  0.66 |    1 |

**Pull request closed without merging**